src/java.base/share/classes/java/lang/doc-files/ValueBased.html

```diff
@@ -31,36 +31,40 @@
 <body>
 <h1 id="ValueBased">{@index "Value-based Classes"}</h1>
 
-Some classes, such as <code>java.util.Optional</code> and
-<code>java.time.LocalDateTime</code>, are <em>value-based</em>.  Instances of a
-value-based class:
+Some classes, such as <code>java.lang.Integer</code> and
+<code>java.time.LocalDate</code>, are <em>value-based</em>.
+Value-based classes:
 <ul>
-    <li>are final and immutable (though may contain references to mutable
-        objects);</li>
-    <li>have implementations of <code>equals</code>,
+    <li>declare only final instance fields (though these may contain references
+        to mutable objects);</li>
+    <li>declare implementations of <code>equals</code>,
         <code>hashCode</code>, and <code>toString</code> which are computed
-        solely from the instance's state and not from its identity or the state
-        of any other object or variable;</li>
-    <li>make no use of identity-sensitive operations such as reference
-        equality (<code>==</code>) between instances, identity hash code of
-        instances, or synchronization on an instances's intrinsic lock;</li>
-    <li>are considered equal solely based on <code>equals()</code>, not
-        based on reference equality (<code>==</code>);</li>
-    <li>do not have accessible constructors, but are instead instantiated
-        through factory methods which make no commitment as to the identity
-        of returned instances;</li>
-    <li>are <em>freely substitutable</em> when equal, meaning that interchanging
-        any two instances <code>x</code> and <code>y</code> that are equal
-        according to <code>equals()</code> in any computation or method
-        invocation should produce no visible change in behavior.
-    </li>
+        solely from the values of the class's instance fields (and properties of
+        the objects they reference), not from the instance's identity</li>
+    <li>have instances that are <em>freely substitutable</em> when equal, meaning
+        that interchanging any two instances <code>x</code> and <code>y</code> that
+        are equal according to <code>equals()</code> produces no visible change in
+        the behavior of the class's methods;</li>
+    <li>perform no synchronization using an instance's monitor;</li>
+    <li>do not have (or have deprecated any) accessible constructors;</li>
+    <li>do not provide any other instance creation mechanism that promises
+        a unique identity on each method call&mdash;in particular, any factory
+        method must allow for the possibility that if two independently-produced
+        instances are equal according to <code>equals()</code>, they may also be
+        equal according to <code>==</code>;</li>
+    <li>are final;</li>
```

Comment:
The way this has been handled until now is to say "all implementing classes should be value-based" in the abstract class's (or factory method's) javadoc. In terms of prose, that seems to get the job done, without needing to complicate the definition of *value-based class*.

For the annotation, yes the intent is that it will be applied both to value-based classes and to abstract classes/interfaces that require all implementing classes to be value-based.